contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -32,7 +32,13 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
 
         bytes32 checkpointHash = keccak256(abi.encode(checkpoint));
 
-        if (checkpoint.blockHeight == s.lastBottomUpCheckpointHeight + s.bottomUpCheckPeriod) {
+        if (checkpoint.blockHeight <= s.lastBottomUpCheckpointHeight) {
+            revert BottomUpCheckpointAlreadySubmitted();
+        }
+
+        // When a bottom up msg batch becomes full, a bottom up checkpoint will be
+        // created before the next checkpoint period is reached.
+        if (checkpoint.blockHeight <= s.lastBottomUpCheckpointHeight + s.bottomUpCheckPeriod) {
```

Comment:
We should throw an error when the checkpoint's block height is higher than expected. Right now we will succeed silently. 

So this method should look like a traditional validate-then-execute pattern.
1. validate checkpoint is not below or equal to latest checkpointed height; if not, error.
2. validate checkpoint is within range; if not, error.

Then perform the logic outside of an if block.